src/Chapter2.hs

```diff
@@ -348,9 +367,12 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf xs
+  | even len = take (div len 2) xs
+  | otherwise = take (div (len-1) 2) xs
```

Comment:
You don't need to check `len` separately, because `div` by two for an odd number and the previous one is the same

Reply:
🤦‍♂️ - yep silly of me 👍